tfjs-backend-webgpu/src/cum_webgpu.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {getMainHeaderAndGlobalIndexString} from './shader_preprocessor';
+import {WebGPUProgram} from './webgpu_program';
+import {computeDispatch, flatDispatchLayout} from './webgpu_util';
+
+export enum CumOpType {
+  Prod = '*',
+  Sum = '+',
+}
+
+export class CumProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x'];
+  workGroupSize: [number, number, number];
+  // pow(f32, i32) is not supported, use pow(f32, f32) instead.
```

Comment:
nit: `pow(i32, i32) is not supported`? Usually, we use the same type.

Reply:
Not support.